### PR TITLE
fix[security]: Disable nginx server signature by default

### PR DESCRIPTION
**Pull request by jackylamhk**
### Motivation and context

Disable nginx server signature to not display nginx and OS version to the public, which will make vulnerability discovery by third-party actors slightly more difficult.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **Security Enhancements**
	- Improved privacy by configuring the server to not disclose software version numbers in responses.
	- Enhanced security by disabling the server signature to make it harder for attackers to exploit known vulnerabilities.

### CODE REVIEW

**Comment by nmanovic** [comment]
@jackylamhk , thank you for the PR. Could you please add a comment with explanations why the change is necessary into the nginx config? Also, please add a line into changelog. Every PR which improves something, should have something like that. Have a great day!

**Comment by jackylamhk** [comment]
@nmanovic Thanks for reviewing this. See comment above for explaination, and I believe `CHANGELOG.md` is auto-updated via GitHub Actions.

**Comment by nmanovic** [comment]
@jackylamhk , I read comments. I want to have the comment in the code to explain it explicitly. Also, CHANGELOG.md isn't updated automatically. See how other PRs do that and also read instructions inside CHANGELOG.md.

**Comment by jackylamhk** [comment]
@nmanovic Sorry - I misunderstood your original request. I've added an in-line code comment to provide context and created the CHANGELOG fragment as requested. Kindly let me know if anything else needs to be done.

File: changelog.d/20240429_124319_jackylamhk_patch_2.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Security
+
+- Disable the nginx server signature by default to make it slightly harder for attackers to find known vulnerabilities. (<https://github.com/cvat-ai/cvat/pull/7814>)
```
Comment (nmanovic) [inline changelog.d/20240429_124319_jackylamhk_patch_2.md:3]:
```suggestion
- Disable the nginx server signature by default to make it slightly harder for attackers to find known vulnerabilities.
  (<https://github.com/cvat-ai/cvat/pull/7814>)
```

**Review by nmanovic — APPROVED** [review]
LGTM, thank you for the contribution. After tests and linters are green, I will merge.

File: changelog.d/20240429_124319_jackylamhk_patch_2.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+### Security
+
+- Disable the nginx server signature by default to make it slightly harder for attackers to find known vulnerabilities.
+  (<https://github.com/cvat-ai/cvat/pull/7814>)
```
[the remark this answers is not in the document]
Comment (nmanovic) [reply changelog.d/20240429_124319_jackylamhk_patch_2.md:4]: We use a template, which we shared with you previously. The link should be on another line. Also, the whole line should be less than 120 characters in accordance with configuration of Remark linter.

[the remark this answers is not in the document]
Comment (nmanovic) [reply changelog.d/20240429_124319_jackylamhk_patch_2.md:4]: OK. It looks exactly as I wrote. Do you have any other comments?

**Comment by jackylamhk** [comment]
@nmanovic Sure - just pushed a new commit.

**Pull request merged**